packages/playground/src/context/messages/reducer.ts

```diff
@@ -0,0 +1,171 @@
+import { CompilationResult } from '@paritytech/commontypes';
+import { Message, Status, Severity, Prompt } from '@paritytech/components/';
+
+export const defaultState: MessageState = {
+  messages: [],
+  nextId: 0,
+};
+
+export type MessageState = {
+  messages: Array<Message>;
+  nextId: number;
+};
+
+export type Action = {
+  type: 'LOG_COMPILE' | 'LOG_SYSTEM' | 'LOG_GIST';
+  payload: {
+    status: Status;
+    content: string;
+    result?: CompilationResult;
+  };
+};
+
+export type MessageDispatch = (action: Action) => void;
+
+const lastId = (state: MessageState, prompt: Prompt): number => {
+  const arr = state.messages.filter(message => message.prompt === prompt);
+  // if no last id available, return nextId, even though it should not happen
```

Comment:
This error handling might be unnecessary, because dispatching a message of state "DONE" should never happen, when the was no "IN_PROGRESS" before. Might be good, to catch/handle the error anyways.

Reply:
@achimcc Should we keep this?

Reply:
sorry to be pedantic here :) but "should never happen".... where is it formalized?

Reply:
It isn't so we better keep the fallback id, I'd say.